NachoSpigot-Server/src/main/java/dev/cobblesword/nachospigot/hitdetection/LagCompensator.java

```diff
@@ -0,0 +1,96 @@
+package dev.cobblesword.nachospigot.hitdetection;
+
+import me.elier.nachospigot.config.NachoConfig;
+import org.bukkit.Location;
+import org.bukkit.entity.Player;
+import org.bukkit.util.Vector;
+
+import java.util.*;
+
+public class LagCompensator {
+
+    /*
+        Lag Compensator based on https://github.com/Islandscout/ServerSideHitDetection
+        Adjusts hit detection to be more fair for players with worse ping.
+     */
+
+    private static final Map<UUID, List<Pair<Location, Long>>> locationTimes = new HashMap<>();
+    private static final int historySize = 35;
+    private static final int pingOffset = 175;
+    private static final int TIME_RESOLUTION = 30;
```

Comment:
constants names should be like that:
HISTORY_SIZE
PING_OFFSET
TIME_RESOLUTION